### PR TITLE
Issue #85: Add the ability to edit existing channels

### PR DESCRIPTION
**Pull request by tomb1n0**
This addresses the edit portion of issue #85.

Not sure on how the deletion should be handled at this stage.

### CODE REVIEW

**Comment by MattMangoni** [comment]
Hmm.. I don't remember if we were using soft deletes or not, but I don't think _that_ alone can solve the problem. We could try to assign all the channel's threads to a default one (or just leave them uncategorized) in order to keep the content available to the users, but that would require us creating a default generic channel on forum installation, something I don't really like.

Or maybe we can just (soft) delete everything. Deleting the PHP channel would defeat the purpose of having PHP related threads, after all.

**Comment by deframe** [comment]
I don't see much point in using soft deletes unless there's an undelete/reactivate feature. IMO someone who deletes a channel will want to get rid of all associated posts rather than reassigning them elsewhere, but perhaps that should be an option of some sort?

**Comment by tomb1n0** [comment]
I'm not sure on removing all associated posts - forums tend to contain useful information so having these threads searchable (but locked maybe?) after the channel is soft deleted would be quite useful to me.
Perhaps a compromise would be to have an option in the frontend (or in the config files) to do one or the other.

**Comment by jonsugar** [comment]
I don't think we should delete existing threads. 

Maybe we could have the option to **_archive_** a channel? Given a channel PHP, archiving would create a new channel named 'PHP (Archived)' and move the existing threads to it. Potentially a new 'archived' flag on the channels table could be used as well if we want to hide archived channels from searches or have other archives specific functionality. After archiving the channel, the original one can be safely deleted.

Maybe a 'move' feature would be useful. Allowing threads to be transferred to another channel during the delete process. Again, the channel could be safely deleted after moving the threads. It all depends on what the forum owner wants—archival of the channel or deletion of the channel but preserving the threads.

File: resources/views/admin/channels/index.blade.php [hunk 1]
```diff
@@ -19,7 +20,10 @@
                     <td>{{$channel->name}}</td>
                     <td>{{$channel->slug}}</td>
                     <td>{{$channel->description}}</td>
-                    <td>{{ $channel->threads()->count() }}</td>
+                    <td>{{$channel->threads()->count()}}</td>
+                    <td>
+                        <a href="/admin/channels/{{ $channel->slug }}/edit" class="btn btn-default btn-xs">Edit</a>
```
Comment (goncalomcunha) [inline resources/views/admin/channels/index.blade.php]: use named route here

**Comment by tomb1n0** [comment]
@JeffreyWay what do you think the best approach to deleting a channel would be?

**Comment by bambamboole** [comment]
In my opinion it would be the easiest to just flag a channel as archived with a column in the table and  filter them out by default.

**Comment by JeffreyWay** [comment]
Agree that toggling an "archived" column is the simplest way to go. I have that done locally, and will push it up tomorrow.

**Pull request merged**